autotest/gcore/geoloc.py

```diff
@@ -600,3 +600,26 @@ def test_geoloc_warnings_inconsistent_size(
         else:
             assert gdal.GetLastErrorMsg() == ""
         assert tr
+
+
+###############################################################################
+# Test effect of normalizing longitude from geolocation arrays
+
+
+@pytest.mark.skipif(
+    not gdaltest.vrt_has_open_support(),
+    reason="VRT driver open missing",
+)
+def test_geoloc_normalize_longitude():
+    vrt = "../gcore/data/geolocation_arrays_lon180_lon360.vrt"
+    warped_ds = gdal.Warp(
+        "",
+        vrt,
+        format="MEM",
+        outputBounds=(-180, -90, 180, 90),
+        width=6,
+        height=3,
+        dstSRS="EPSG:4326",
+        transformerOptions=["GEOLOC_NORMALIZE_LONGITUDE_MINUS_180_PLUS_180=YES"],
+    )
+    assert warped_ds.ReadRaster(2, 0, 1, 1) == bytearray(b"\x00\x00@A")
```

Comment:
would be more readable as:
```suggestion
    assert struct.unpack('f', warped_ds.ReadRaster(2, 0, 1, 1)) == the_value
```